### PR TITLE
 feat: add support for ARM architecture

### PR DESCRIPTION
**Pull request by reschandreas**
During my Artemis development I noticed that I could not create/manage any haskell exercises due to my computer's architecture. This PR adds a QEMU step and publishes images for amd and arm.

The QEMU build takes about 1h and 20 minutes

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -8,10 +8,10 @@ COPY stack-config/config.yaml $STACK_ROOT/config.yaml
...
 
 RUN apt-get update \
-    && apt-get -y install curl perl tar \
-    && bash -c "curl -sSL https://get.haskellstack.org/ | sh" \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/*
+  && apt-get -y install curl perl tar llvm libnuma-dev \
+  && bash -c "curl -sSL https://get.haskellstack.org/ | sh" \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
```
Comment (b-fein) [inline Dockerfile:14]:
```suggestion
    && apt-get -y install curl perl tar llvm libnuma-dev \
    && bash -c "curl -sSL https://get.haskellstack.org/ | sh" \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*
```

File: Dockerfile [hunk 2]
```diff
@@ -27,4 +27,4 @@ RUN stack setup \
             -- --version \
     # Jenkins runs the builds not as root, but with a system dependent user id.
     # Therefore, we allow all users to access the stack cache.
-    && chmod -R a+rw $STACK_ROOT
+    && chmod -R a+rw $STACK_ROOT
```
Comment (b-fein) [inline Dockerfile:30]:
```suggestion
    && chmod -R a+rw $STACK_ROOT

```
Missing newline at end of file (I think)

File: Dockerfile [hunk 3]
```diff
@@ -8,10 +8,10 @@ COPY stack-config/config.yaml $STACK_ROOT/config.yaml
 COPY stack-config/global-project/stack.yaml $STACK_ROOT/global-project/stack.yaml
 
 RUN apt-get update \
-    && apt-get -y install curl perl tar \
-    && bash -c "curl -sSL https://get.haskellstack.org/ | sh" \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/*
+  && apt-get -y install curl perl tar llvm libnuma-dev \
```
Comment (b-fein) [inline Dockerfile:11]: What are those needed for?

**Review by b-fein — APPROVED** [review]
Thanks for the contribution!

Only two formatting nitpicks, and one question. Otherwise looks good if the pipeline passes.

---

Note to self: I’ll have to re-tag on `main` (or update to a newer Stack version here and in Artemis) to publish a new image that supports both architectures.

File: Dockerfile (hunk 3 above)
Comment (reschandreas) [reply Dockerfile:11]: I tested it on my laptop and without llvm the docker image can't be built. I think llvm is used on arm to enhance the code generation.

**Comment by reschandreas** [comment]
@b-fein thanks for approving, sometimes the pipeline fails because it takes too long and GitHub thinks it has lost connection. Just rerunning the job worked for me if I saw `The hosted runner: GitHub Actions 2 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.`

**Comment by reschandreas** [comment]
The action keeps failing because GitHub kills it, if GitHub would support ARM runners (or you have a self-hosted one), the arm image would build, just check my fork, it worked there.

File: .github/workflows/docker-publish.yml [hunk 4]
```diff
@@ -19,10 +19,8 @@ env:
   # github.repository as <account>/<repo>
   IMAGE_NAME: ${{ github.repository }}
 
-
 jobs:
   build:
-
     runs-on: ubuntu-latest
```
Comment (b-fein) [inline .github/workflows/docker-publish.yml:24]:
```suggestion
    runs-on: self-hosted
```

Comment (reschandreas) [reply .github/workflows/docker-publish.yml]: perfect, thanks

**Pull request merged**